docs: document expected behaviour on user delete (EDP) (#2529)
TykTechnologies/tyk-docs: portal/api-owner.mdx

4. Confirm the deletion
5. All user data will be permanently removed

> When a portal/super-admin performs the same action, both the user and their apps (and credentials) are permanently deleted.

Reviewer: ```suggestion
When an API Owner performs the delete action, both the user and their apps (and credentials) are permanently deleted.
```
